### PR TITLE
feat(analytics): add setConsent implementation

### PR DESCRIPTION
**Pull request by Summys**
### Description

A well put description can be found at https://github.com/invertase/react-native-firebase/discussions/6636

The motivation for this change is coming from Google where we have to ask for end user consent by 6th March. 

Relevant URLs:

- [[GA4] Verify and update consent settings in Google Analytics](https://support.google.com/analytics/answer/14275483)
- [Updates to consent mode for traffic in European Economic Area (EEA)](https://support.google.com/tagmanager/answer/13695607)

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [x] Yes
- My change supports the following platforms;
  - [x] `Android`
  - [x] `iOS`
- My change includes tests;
  - [x] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [x] `jest` tests added or updated in `packages/\*\*/__tests__`
- [x] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [x] No

### Test Plan

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

File: packages/analytics/lib/modular/index.js [hunk 1]
```diff
@@ -652,11 +652,11 @@ export function isSupported() {
  * references once Firebase Analytics is initialized. Web only.
```
Comment (mikehardy) [inline packages/analytics/lib/modular/index.js:652]: comment here needs updating as well - the update in index.d.ts looked perfect

File: packages/analytics/__tests__/analytics.test.ts [hunk 2]
```diff
@@ -98,6 +98,31 @@ describe('Analytics', function () {
     );
   });
 
+  it('throws if consentSettings is not an object', function () {
```
Comment (mikehardy) [inline packages/analytics/__tests__/analytics.test.ts:101]: If I had to prefer one spot - the testing should be on the modular APIs (which is likely a nearly verbatim hunk of tests in the second half of this file ? - but testing in both is great. What's missing either way is the modular API test though

**Review by mikehardy — CHANGES_REQUESTED** [review]
This looks really thorough, *thank you*

I left two comments which are both (in my opinion at least) trivial to fix - duplicating the test you added but using the modular APIs in the modular section of the e2e test file and updating the method docs in one spot to match the update you made in both

Going to approve the CI runs now as well, they always seem to turn up some trivial spellcheck or lint issue as well

...but in general this looks 99.9% ready to go

**Comment by Summys** [comment]
Added e2e tests and fixed linting errors. Thanks for your quick review @mikehardy

**Review by mikehardy — APPROVED** [review]
This looks really great. Thank you!
I'll start shepherding it through CI
I'm interactive on the repo right now so I'm fine fixing any lint errors or spelling things etc that pop up (there always seems to be something, and always really easy to fix)
Should merge today and I'll queue it for release shortly

The patch set should be useful already either way if you follow the details link and then look at [the summary on the patch-package action](https://github.com/invertase/react-native-firebase/actions/runs/7958134946?pr=7629)

**Comment by nixolas1** [comment]
Great stuff, thanks for the quick work both of you :)
Will be testing the patch tomorrow.

**Comment by mikehardy** [comment]
Something isn't quite right in the gradle changes but they looked good on visual inspection so it can't be far off.

Hold off on using the patch set of course, but I'll fix up whatever little thing isn't right and get this in

**Comment by Summys** [comment]
> Something isn't quite right in the gradle changes but they looked good on visual inspection so it can't be far off.
> 
> Hold off on using the patch set of course, but I'll fix up whatever little thing isn't right and get this in

the variables I added weren't defined in gradle

**Review by mikehardy — APPROVED** [review]
😆 that'll do it, CI chewing on things now

**Comment by mikehardy** [comment]
One of the android APIs (23) wasn't happy but one passed - that's definitely sufficient for the feature as the e2e tests are a bit flaky

**Comment by mikehardy** [comment]
For those looking for a patch set - this was the workflow run with patch-set from the merge to main: https://github.com/invertase/react-native-firebase/actions/runs/7963378973

**Comment by RamProg** [comment]
How can I use this commit before the new release?

**Comment by nixolas1** [comment]
Download the patches folder https://github.com/invertase/react-native-firebase/actions/runs/7963378973/artifacts/1257279870 and put it in your root, make sure youre on 18.8.0, and install patch-package. When you run yarn install it will patch the files with the new changes.

**Comment by RamProg** [comment]
Yes, thank you, I was able to do that. Do I need the whole patches folder or just the analytics patch (I'm only interested in this particular change).

**Comment by nixolas1** [comment]
All the packages you have installed. It'l give you a patch error if you have patches for non-installed packages.

**Comment by RamProg** [comment]
I have them all installed, but only care about changes in Analytics, I still need to apply them all right?

**Comment by Ace090** [comment]
I have an older application that uses versions
```
"@react-native-firebase/analytics": "^11.5.0",
"@react-native-firebase/app": "^11.5.0",
"@react-native-firebase/messaging": "^11.5.0",
```

How can I use `setContent` in this version?

**Pull request merged**